diff --git a/nexus/Freqlog/Freqlog.py b/nexus/Freqlog/Freqlog.py
--- a/nexus/Freqlog/Freqlog.py
+++ b/nexus/Freqlog/Freqlog.py
@@ -212,17 +212,9 @@
                     self.q.task_done()
                     continue

                 banned_modifier_active = False
-                for attr in Defaults.MODIFIER_NAMES:
-                    # Skip non-enabled modifiers
-                    if not getattr(self.modifier_keys, attr):
-                        continue

-                    if getattr(modifiers, attr):
-                        banned_modifier_active = True
-                        break
-
                 # Add new char to word and update word timing if no banned modifier keys are pressed
                 if not banned_modifier_active:
                     # I think this is for chords that end in space
                     # If last key was disallowed and timing of this key is more than chord_char_threshold, log+reset
